packages/admin/admin/src/mui/MainContent.tsx

```diff
@@ -17,20 +19,53 @@ const styles = ({ spacing }: Theme) => {
             zIndex: 5,
             padding: spacing(4),
         },
+        fullHeight: {
+            height: "calc(100vh - var(--comet-admin-main-content-top-position))",
+        },
         disablePaddingTop: {
             paddingTop: 0,
         },
         disablePaddingBottom: {
             paddingBottom: 0,
         },
+        disablePadding: {
+            padding: 0,
+        },
     });
 };
 
-function Main({ children, disablePaddingTop, disablePaddingBottom, classes }: MainContentProps & WithStyles<typeof styles>) {
+function Main({
+    children,
+    fullHeight,
+    disablePaddingTop,
+    disablePaddingBottom,
+    disablePadding,
+    classes,
+}: MainContentProps & WithStyles<typeof styles>) {
+    const mainRef = React.useRef<HTMLElement>(null);
+    const [topPosition, setTopPosition] = React.useState<number>(0);
+
     const rootClasses: string[] = [classes.root];
     if (disablePaddingTop) rootClasses.push(classes.disablePaddingTop);
     if (disablePaddingBottom) rootClasses.push(classes.disablePaddingBottom);
-    return <main className={rootClasses.join(" ")}>{children}</main>;
+    if (disablePadding) rootClasses.push(classes.disablePadding);
+    if (fullHeight) rootClasses.push(classes.fullHeight);
+
+    React.useEffect(() => {
+        if (fullHeight && mainRef.current) {
+            setTopPosition(mainRef.current.offsetTop);
+        }
+    }, [fullHeight]);
+
+    return (
+        <main
+            ref={mainRef}
+            className={rootClasses.join(" ")}
+            style={{ "--comet-admin-main-content-top-position": `${topPosition}px` } as React.CSSProperties}
```

Comment:
Shouldn't this be possible without effect and state?

```suggestion
            style={{ "--comet-admin-main-content-top-position": `${fullHeight && mainRef.current ? mainRef.current.offsetTop : 0}px` } as React.CSSProperties}
```